config/config.go

```diff
@@ -518,7 +515,15 @@ func (c WorkerSettings) MaxWorkerCount() int {
 
 func (c WorkerSettings) WorkerRollingUpdateMinInstancesInService() int {
 	if c.AutoScalingGroup.RollingUpdateMinInstancesInService == 0 {
-		return c.MaxWorkerCount() - 1
+
+		maxWorkerCount := c.MaxWorkerCount()
+
+		//never accept worker size = 0
+		if maxWorkerCount > 0 {
+			return maxWorkerCount - 1
+		}
+
+		return 1
```

Comment:
Fixed the min instance for rolling update in the NewDefaultCluster() to 1. If someone doesn't like this value can easily change on cluster.yaml. 

Why this default value? If you don't specify this option and you have only 1 worker/controller you'll have a downtime when upgrading your cluster.
With this option, UpdateMinInstancesInService, you could setup your minsize and maxsize to 0 without problems.